File: DSA/Hima-Swarupa-Padala/day1q1.py
def birthday(candy, day, month):
    count = 0
    for i in range(len(candy) - month + 1):
        if sum(candy[i:i+month]) == day:
            count += 1
    return count

candy = [4, 5 ,4, 1, 2, 4, 3]
day = 9
month = 3
print(birthday(candy, day, month)) 

File: DSA/Yash_Gupta/Day_1_Q1.java
public class Day_1_Q1 {
    public static int countWays(int []candy,int day,int month){
        int count=0;
        for(int i=0;i<=candy.length-month;i++){
            int sum=0;
            for(int j=0;j<month;j++){
                sum+= candy[i+j];
            }
            if(sum==day){
                count++;
            }
        }
        return count;
    }
    public static void main(String[] args) {
        int candy_case_1[]={2, 2, 1, 3, 2};
        int birth_day=4;
        int birth_month=2;
        System.out.println(countWays(candy_case_1,birth_day,birth_month));

        int candy_case_2[]={2, 2, 1, 3, 2};
        birth_day=2;
        birth_month=5;
        System.out.println(countWays(candy_case_2,birth_day,birth_month));

    }
}
